docs/recipes/gulp/gulpfile.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const gulp = require('gulp');
+const connect = require('gulp-connect');
+const lighthouse = require('lighthouse');
+const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher').ChromeLauncher;
+const perfConfig = require('lighthouse/lighthouse-core/config/perf.json');
+const PORT = 8080;
+let launcher;
+
+/**
+ * Connect to server
+ */
+const connectServer = function() {
+  return connect.server({
+    root: './public',
+    livereload: true,
+    port: PORT
+  });
+};
+
+/**
+ * Disconnect server
+ */
+const disconnectServer = function() {
+  connect.serverClose();
+  launcher.kill();
+};
+
+/**
+ * Launch chrome
+ */
+const launchChrome = function() {
+  launcher = new ChromeLauncher();
+  return launcher.isDebuggerReady()
+    .catch(() => {
+      return launcher.run();
+    });
+};
+
+/**
+ * Run lighthouse
+ */
+const runLighthouse = function() {
+  const url = `http://localhost:${PORT}/index.html`;
+  const lighthouseOptions = {}; // available options - https://github.com/GoogleChrome/lighthouse/#cli-options
+  return lighthouse(url, lighthouseOptions, perfConfig)
+    .then(handleOk)
+    .catch(handleError);
```

Comment:
also a nit: should `.then(handleOk).catch(handleError);` be moved into the `gulp.task('lighthouse')` block? So it would be
```js
gulp.task('lighthouse', function() {
  launchChrome().then(_ => {
    connectServer();
    return runLighthouse()
      .then(handleOk);
  }).catch(handleError);
});
```
or something? Seems to make more sense there than in `runLighthouse`